### PR TITLE
196  - Implement my dataset download email form in UI using useDatasetManager

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number

Closing #196

- This branch was created based on [#199](https://github.com/AlexsLemonade/refinebio-web/pull/199) (will be merged first).
- This PR's commit starts at 695df112d57c43dff49b2798d9f6dc76597d7d84

## Purpose/Implementation Notes
Implemented the my dataset download email form in UI using useDatasetManager.

Please see the latest UI [here](https://refinebio-12qtegal2-ccdl.vercel.app/).

To test the feature, please use the following dataset Id or use any datasets you've created (the URL path should be `VERCEL_PREVIEW_URL/dataset/DATAET_ID`):

An expired dataset Id `43150960-ccd4-49ac-9c82-f6fad8ab12a9`:
https://refinebio-12qtegal2-ccdl.vercel.app/dataset/43150960-ccd4-49ac-9c82-f6fad8ab12a9

**Expected behavior:** Please regenerate the dataset using the download options form in the "Download Files Summary" section (to toggle the form, click the 'Change' link-button). In the download email form, after entering a valid email address and agreeing to the terms of use, click the 'Start Processing' button to start the dataset processing. Once the file download is ready, you'll receive an email with a download link (the same flow for unexpired datasets or datasets in users' "My Dataset").

The application will auto-fill the email address field in the download email form if a user previously provided their email address.

(it matches the behavior of the current refine.bio)

## Types of changes
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by davidsmejia — COMMENTED** [review]
Please rename the `EmailForm` to `CreateTokenForm` to help clarify it's purpose, also update the validation schema's name to match.

**Comment by nozomione** [comment]
> Please rename the `EmailForm` to `CreateTokenForm` to help clarify it's purpose, also update the validation schema's name to match.

This form is for Hotjar email subscriptions, and before starting processing a dataset, a user must enter a valid email address and accept the terms of use (by agreeing to the terms, a token will be generated to verify that user for the file download)

In this context, the name `EmailForm` is suitable so, please let us keep this name as is.

However, in the later PR https://github.com/AlexsLemonade/refinebio-web/pull/206 (for the shared dataset download modal), those fields were split into sub-components (eeadbfbaf4a2b34c5598a098eae7b7caa6a03b32) for reusability. In order to implement your suggestion, may I rename the  [`TermOfUseCheckBox`](https://github.com/AlexsLemonade/refinebio-web/pull/206/commits/eeadbfbaf4a2b34c5598a098eae7b7caa6a03b32#diff-e2c51d983f0fac2ce9faaeb3fafc93ecdb6c812f2be4d83ba9008541bba86f8a) component (must be checked for generating a token) to `CreateTokenCheckbox` in this later PR?

**Comment by davidsmejia** [comment]
> This form is for Hotjar email subscriptions, and before starting processing a dataset, a user must enter a valid email address and accept the terms of use (by agreeing to the terms, a token will be generated to verify that user for the file download)

Hmm, I don't think I would look at this form and say that it is here for hotjar. In my opinion this form is for accepting the terms and conditions. While we can reuse a previously generated token here, for legal reasons we must ask them to accept the terms as they are at the time of download.

Either way, calling it `EmailForm` is not what I would expect. Since the function of the form we should just name it based off of what it does. So lets instead call in `StartProcessingForm` since on submit the purpose is to update the dataset to kick off processing.

**Comment by nozomione** [comment]
> Hmm, I don't think I would look at this form and say that it is here for hotjar. In my opinion this form is for accepting the terms and conditions. While we can reuse a previously generated token here, for legal reasons we must ask them to accept the terms as they are at the time of download.
> 
> Either way, calling it `EmailForm` is not what I would expect. Since the function of the form we should just name it based off of what it does. So lets instead call in `StartProcessingForm` since on submit the purpose is to update the dataset to kick off processing.

This form is responsible for the following: 
- Collecting users' valid email address 
- Signing users up for Hotjar email subscription if its checkbox is selected
- Enforcing users to accept the terms of use 

So it is reasonable to call it as `EmailForm`, but `StartProcessingForm` is also a good name 👍 

As I mentioned, this  `EmailForm` components will be broken up into sub-components for reusability. 

> ...in the later PR https://github.com/AlexsLemonade/refinebio-web/pull/206 (for the shared dataset download modal), those fields were split into sub-components (eeadbfbaf4a2b34c5598a098eae7b7caa6a03b32) for reusability. 

So I'll go ahead rename this to `StartProcessingForm  ` in PR https://github.com/AlexsLemonade/refinebio-web/pull/206, thank you David!

**Comment by nozomione** [comment]
@davidsmejia, I've rename the component and its validation schema in PR #206 (please see the commits: a260f5cb0370528ef7dadf29108d616848b3f646 ~ 88d5b1969aaf292ce1dffb29ca6d74351e7fbd94) ✅ and this PR is ready for another look.

File: src/components/Download/EmailForm.js [hunk 1]
```diff
@@ -1,52 +1,131 @@
...
+              {errors.emailAddress && touched.emailAddress && (
+                <Box animation={{ type: 'fadeIn', duration: 300 }}>
+                  <InlineMessage
+                    color="error"
+                    height="16px"
+                    justify="center"
+                    label={errors.emailAddress}
+                    iconSize="small"
+                    style={{ position: 'absolute', top: '-20px' }}
+                  />
```
[the remark this answers is not in the document]
Comment (nozomione) [reply src/components/Download/EmailForm.js:65]: Sure (I liked that design too - it was nice 🎨!) Once again, I broken up this `EmailForm` component (now renamed as `StartProcessingForm`) into sub-components for usability in PR https://github.com/AlexsLemonade/refinebio-web/pull/206, so I'll go ahead make these changes in PR https://github.com/AlexsLemonade/refinebio-web/pull/206, thank you David!

**Review by davidsmejia — APPROVED** [review]
This LGTM as we will address my concerns in referenced PRs.

**Pull request merged**